torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -126,38 +128,119 @@ def auto_functionalized_dense(
     _only_clone_these_tensors: Optional[Tuple[str, ...]] = None,
     **kwargs: Any,
 ) -> Tuple[Any, Tuple[Tensor, ...]]:
+    _all_bases: List[Tensor] = kwargs.pop("_all_bases", [])
```

Comment:
I just realized we need to handle the following case in this PR: if the base has no view relationship with the tensor (maybe due to a graph pass), then we don't know what the right thing to do is.

So, in addition to recording x_base for each x, we also need to record x.stride() and x.storage_offset(). This needs to happen in this PR (sorry I didn't realize earlier)

Reply:
no worries lets chat about this and about the change that i made to this. to handle test_multi_output_intermediate when you have time. I would like to understand better this comment.

Reply:
Under this design we wouldn't need to pass `x` anymore (it doesn't matter what's in x!). Instead we can pass (x_sizes, x_strides, x_storage_offset) instead of x.

Also, the semantics of auto_functionalized become:

```py
x_base = maybe_clone(x_base)  # if we can't reinplace it
x = x_base.as_strided(x_sizes, x_strides, x_storage_offset)
mutable_op(x)
```
But we also want to somehow not emit an as_strided when it's unnecessary